analyzers/src/SonarAnalyzer.Common/Json/JsonWalker.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Json.Parsing;
+
+namespace SonarAnalyzer.Json
+{
+    public abstract class JsonWalker
+    {
```

Comment:
Fix the code smell https://sonarcloud.io/project/issues?pullRequest=6201&issues=AYPLr8WRbucSp08jW4_H&open=AYPLr8WRbucSp08jW4_H&id=sonaranalyzer-dotnet

```suggestion
    public class JsonWalker
    {
        protected JsonWalker() { }

```

Reply:
I have very mixed feeling about this suggestion. It IMO just moves the "inherit me" (aka `MustInherit` in VB) information somewhere deep in the code, making it less readable.

Reply:
Same here. And the description of the rule has a flaw too:
https://github.com/SonarSource/rspec/pull/1341